state/keys.go

```diff
@@ -59,13 +62,15 @@ func (k Keys) ChunkSizes() ([]uint16, bool) {
 type permsJSON []string
 
 type keysJSON struct {
-	Perms [8]permsJSON
+	Perms map[string]permsJSON
```

Comment:
Can we change the key type to `codec.Bytes` so that it uses the standard serialization throughout the repo for encoding the binary key values?

Once we do this, we can replace the existing `keysJSON` and `permsJSON` types with a single intermediate type `map[codec.Bytes]Permissions` and call `json.Marshal` / `json.Unmarshal` on the intermediate type